src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -693,6 +762,128 @@ export class StudyViewPageStore {
         },
     });
 
+    readonly userSavedCustomCharts = remoteData<
+        (ChartMeta & {
+            data: CustomChartIdentifierWithValue[];
+            isSharedChart?: boolean;
+        })[]
+    >({
+        invoke: async () => {
+            if (this.studyIds.length > 0 && this.isLoggedIn) {
+                const chustomChartsSessions = await sessionServiceClient.getCustomDataForStudies(
+                    this.studyIds.slice()
+                ); // slice because cant pass mobx
+
+                return chustomChartsSessions.map(chustomChartsSession => {
+                    return {
+                        uniqueKey: chustomChartsSession.id,
+                        displayName: chustomChartsSession.data.displayName,
+                        description: chustomChartsSession.data.description,
+                        priority: chustomChartsSession.data.priority,
+                        dataType: ChartMetaDataTypeEnum.CUSTOM_DATA,
+                        patientAttribute:
+                            chustomChartsSession.data.patientAttribute,
+                        renderWhenDataChange: false,
+                        data: chustomChartsSession.data.data,
+                        isSharedChart: false,
+                        clinicalAttribute: {
+                            clinicalAttributeId: chustomChartsSession.id,
+                            datatype: chustomChartsSession.data.datatype,
+                            displayName: chustomChartsSession.data.displayName,
+                            description: chustomChartsSession.data.description,
+                            patientAttribute:
+                                chustomChartsSession.data.patientAttribute,
+                            priority: chustomChartsSession.data.priority,
+                        } as any,
+                    };
+                });
+            }
+            return [];
+        },
+        default: [],
+        onResult: customDataList => {
+            if (this._loadUserSettingsInitially) {
+                customDataList.forEach(customChart => {
+                    const uniqueKey = customChart.uniqueKey;
+                    if (!this.customChartSet.has(uniqueKey)) {
+                        let { data, isSharedChart, ...chartMeta } = customChart;
+                        this._customCharts.set(uniqueKey, chartMeta);
+                        this._customChartsSelectedCases.set(uniqueKey, data);
+                        this.chartsType.set(uniqueKey, ChartTypeEnum.PIE_CHART);
+                        this.chartsDimension.set(uniqueKey, { w: 1, h: 1 });
+                    }
+                    this.customChartSet.set(uniqueKey, customChart);
+                });
+            }
+        },
+    });
+
+    readonly sharedCustomCharts = remoteData<
```

Comment:
we need a comment here explaining what "shared" means.   i mean, obviously it means shared from another user, but also that (if i understand correctly) different things will happen with a shared group depending on logged in state of user.  we need to have all this explained at the very least in comments